README.md

```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
 
 ## Development
 
-See the section on pluggable providers in the [ManageIQ Developer Setup](http://manageiq.org/docs/guides/developer_setup)
+This is a quick guide outlining all the necessary steps to get ManageIQ up and
+running for development.
 
-For quick local setup run `bin/setup`, which will clone the core ManageIQ repository under the *spec* directory and setup necessary config files. If you have already cloned it, you can run `bin/update` to bring the core ManageIQ code up to date.
+### Dependencies
+
+* Ruby 2.3+ (Ruby 2.4.1 recommended)
+* Bundler 1.15.3+
+* PostgreSQL 9.5+
+* Memcached
+* Node/npm and the usual suspects: Bower, Yarn, Gulp, Webpack
+* Docker
+* 16gb RAM (8gb or less is doable, but not recommended)
+
+Any Linux/OS X System is compatible, as far as we are concerned.
+
+### Installing Dependencies
+
+#### Fedora
+
+First, install all system dependencies:
+
+```bash
+sudo dnf -y git-all memcached postgresql-devel postgresql-server postgresql-contrib bzip2 libffi-devel readline-devel sqlite-devel nodejs gcc-c++ libcurl-devel npm openssl-devel cmake openscap
+```
+
+Then, install npm packages required for the UI:
+
+```bash
+sudo npm install -g bower yarn webpack gulp
+```
+
+Start Memcached:
+
+```bash
+sudo systemctl enable memcached
+sudo systemctl start memcached
+```
+
+Then, configure, install and run PostgreSQL:
+
+```bash
+sudo postgresql-setup --initdb --unit postgresql
+sudo grep -q '^local\s' /var/lib/pgsql/data/pg_hba.conf || echo "local all all trust" | sudo tee -a /var/lib/pgsql/data/pg_hba.conf
+sudo sed -i.bak 's/\(^local\s*\w*\s*\w*\s*\)\(peer$\)/\1trust/' /var/lib/pgsql/data/pg_hba.conf
+sudo systemctl enable postgresql
+sudo systemctl start postgresql
+sudo su postgres -c "psql -c \"CREATE ROLE root SUPERUSER LOGIN PASSWORD 'smartvm'\""
+```
+
+#### OS X
+
+TODO.
+
+#### Installing a Ruby Runtime
+
+There are many options to have different ruby versions running at the same
+time, and while the Ruby versions in the repositories are usually up to date
+enough, there's often a need to test or fix stuff in different ruby versions,
+so the recommended way is to use a ruby version manager, such as
+[RVM](https://rvm.io/), [Rbenv](https://github.com/rbenv/rbenv) or
+[Chruby](https://github.com/postmodern/chruby). In this guide we'll be covering
+Rbenv.
+
+For OS X, `rbenv` and `ruby-build` are available on homebrew, and can be installed like this:
+
+```bash
+brew upgrade rbenv ruby-build
+echo 'eval "$(rbenv init -)"' >> ~/.bashrc
+eval "$(rbenv init -)"
+```
+
+For Linux, it will have to be installed via default git checkout:
```

Comment:
isn't it easier with rvm?

Reply:
I'm not sure, I think is matter of choice. I use `chruby`. 
If rvm setup is easier we should provide that one here instead, unless there is a real advantage of using rbenv.

Reply:
@cfcosta ^^

Reply:
There is. Afaik, if that didn't change, rvm overwrites system commands like cd and is generally a pain to work with, requiring implodes every now and then. I've been using rbenv since and the workflow couldn't be nicer.

Either way, I'm going to remove installation instructions and let everyone chooses what they want to use.